haystack/api/controller/model.py

```diff
@@ -93,39 +93,39 @@ class Answers(BaseModel):
 #############################################
 # Endpoints
 #############################################
+doc_qa_limiter = RequestLimiter(CONCURRENT_REQUEST_PER_WORKER)
+
 @router.post("/models/{model_id}/doc-qa", response_model=Answers, response_model_exclude_unset=True)
 def doc_qa(model_id: int, request: Question):
-    t1 = time.time()
-    finder = FINDERS.get(model_id, None)
-    if not finder:
-        raise HTTPException(
-            status_code=404, detail=f"Couldn't get Finder with ID {model_id}. Available IDs: {list(FINDERS.keys())}"
-        )
-
-    results = []
-    for question in request.questions:
-        if request.filters:
-            # put filter values into a list and remove filters with null value
-            request.filters = {key: [value] for key, value in request.filters.items() if value is not None}
-            logger.info(f" [{datetime.now()}] Request: {request}")
-
-        result = finder.get_answers(
-            question=question,
-            top_k_retriever=request.top_k_retriever,
-            top_k_reader=request.top_k_reader,
-            filters=request.filters,
-        )
-        results.append(result)
-
-        resp_time = round(time.time() - t1, 2)
-        logger.info({"time": resp_time, "request": request.json(), "results": results})
+    with doc_qa_limiter.run():
```

Comment:
It would be nicer to make a decorator out of this, but if this requires bigger changes we can do it later in a separate PR.